events/rancher_state_watcher.go

```diff
@@ -0,0 +1,172 @@
+package events
+
+import (
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/fsouza/go-dockerclient"
+	"gopkg.in/fsnotify.v1"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"time"
+)
+
+const healthCheckFileName = ".healthcheck"
+
+var healthCheckTimeout = time.Second * 10
+var healthCheckWriteInterval = time.Second * 8
+var restartWaitUnit = time.Second
+var maxRestarts = 5
+
+func watchRancherState(eventChannel chan<- *docker.APIEvents, watchDir string) {
+	// Ideally, I'd prefer to just bubble an error all the way up  with no restart logic
+	// until it causes the process to exit (and be restared)t, but since this code currently
+	// lives with the rest of host-api, I don't feel comfortable being so cavalier with process exits.
+	// So this will try to restart a number of times, waiting expontentially
+	restarts := 0
+	restartWait := 0
+	log.Infof("Watching state directory: %v", watchDir)
+	for {
+		if restarts >= maxRestarts {
+			panic("Unable to successfully start rancher state watcher.")
+		}
+		if err := watchRancherStateInternal(eventChannel, watchDir); err != nil {
+			time.Sleep(restartWaitUnit * time.Duration(restartWait))
```

Comment:
Initial wait time is going to be zero.  We should also log the warning prior to the sleep.